test/hotspot/jtreg/gc/arguments/TestUseCompressedOopsFlagsWithUlimit.java

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright (c) 2022, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+package gc.arguments;
+
+/*
+ * @test TestUseCompressedOopsFlagsWithUlimit
+ * @bug 8280761
+ * @summary Verify correct UseCompressedOops when MaxRAM and MaxRAMPercentage
+ * are specified with ulimit -v.
+ * @library /test/lib
+ * @library /
+ * @requires vm.bits == "64"
+ * @requires os.family != "aix" & os.family != "windows"
+ * @run driver gc.arguments.TestUseCompressedOopsFlagsWithUlimit
+ */
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+
+import jdk.test.lib.process.OutputAnalyzer;
+import jdk.test.lib.process.ProcessTools;
+
+public class TestUseCompressedOopsFlagsWithUlimit {
+
+  private static void checkFlag(long maxram, int maxrampercent, boolean forcecoop, boolean expectcoop) throws Exception {
+
+    ArrayList<String> args = new ArrayList<String>();
+    args.add("-XX:MaxRAM=" + maxram);
+    args.add("-XX:MaxRAMPercentage=" + maxrampercent);
+    if (forcecoop) {
+      args.add("-XX:+UseCompressedOops");
+    }
+
+    args.add("-XX:+PrintFlagsFinal");
+    args.add("-version");
+
+    String cmd = ProcessTools.getCommandLine(ProcessTools.createTestJvm(args.toArray(String[]::new)));
+    ProcessBuilder pb = new ProcessBuilder("sh", "-c", "ulimit -v 10485760;" + cmd);
```

Comment:
The `ulimit -v` parameter should be passed in as well like `MaxRAM`. Then, from the calling context, one can see whether coop should be enabled or not.

Reply:
Thank you for your review.

> I don't understand why `MaxRAMPercentage` or `UseCompressedOops` are relevant to this fix/bug.

I forgot to remove this comment when I copied form TestMaxRAMFlags.java.
I fixed this comment.

> The `ulimit -v` parameter should be passed in as well like `MaxRAM`. Then, from the calling context, one can see whether coop should be enabled or not.

This bug appears when ulimit -v is specified and MaxRAM is set bigger value. So I think it does not need that
ulimit -v parameter let be variable.

Reply:
It's not about whether it's constant or variable.

In this context, it's unclear why coop is enabled.

```java
// Args: MaxRAM , MaxRAMPercentage, forcecoop, expect coop
checkFlag(32 * oneG, 100, false, true);
```

`checkFlag` should not sneak in logic altering the coop value.